skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.copy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+        self.assertIsNot(obj.interval_metadata,
+                         obj_copy.interval_metadata)
+        self.assertIsNot(obj.interval_metadata._intervals,
+                         obj_copy.interval_metadata._intervals)
+        for i, j in zip(obj.interval_metadata._intervals,
+                        obj_copy.interval_metadata._intervals):
+            self.assertIsNot(i, j)
+            self.assertIsNot(i.metadata, j.metadata)
+            for k in i.metadata:
+                self.assertIs(i.metadata[k], j.metadata[k])
+
+    def test_deepcopy_interval_metadata(self):
+        self.im.add(**self.intvls[1])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.deepcopy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+        self.assertIsNot(obj.interval_metadata,
+                         obj_copy.interval_metadata)
+        self.assertIsNot(obj.interval_metadata._intervals,
+                         obj_copy.interval_metadata._intervals)
+        for i, j in zip(obj.interval_metadata._intervals,
+                        obj_copy.interval_metadata._intervals):
+            self.assertIsNot(i, j)
+            self.assertIsNot(i.metadata, j.metadata)
+            self.assertIsNot(i.metadata['gene'], j.metadata['gene'])
+            self.assertIs(i.metadata['product'], j.metadata['product'])
+
+    def test_deepcopy_interval_metadata_empty(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.deepcopy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+    def test_deepcopy_memo_is_respected(self):
+        # Basic test to ensure deepcopy's memo is passed through to recursive
+        # deepcopy calls.
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        memo = {}
+        copy.deepcopy(obj, memo)
+        self.assertGreater(len(memo), 1)
+
+    def test_interval_metadata_getter(self):
+        self.im.add(**self.intvls[0])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        self.assertIsInstance(obj.interval_metadata, IntervalMetadata)
+        self.assertEqual(self.im, obj.interval_metadata)
+
+        # Update existing column.
+        obj.interval_metadata._intervals[0].metadata['gene'] = 'sagB'
+        self.assertNotEqual(obj.interval_metadata, self.im)
+        self.im._intervals[0].metadata['gene'] = 'sagB'
+        self.assertEqual(obj.interval_metadata, self.im)
+
+        # Add new interval feature.
+        obj.interval_metadata.add(**self.intvls[1])
+        self.im.add(**self.intvls[1])
+        self.assertEqual(obj.interval_metadata, self.im)
+
```

Comment:
Also add a test case that asserts the setter works with valid IntervalMetadata. It's being implicitly tested in some of these other test cases but it's good to have an explicit test for this, in case the other tests change in the future. e.g. similar to `test_positional_metadata_setter_data_frame`